### PR TITLE
api: ignore /api/gnet proxy when gzipping responses

### PR DESCRIPTION
**Pull request by sd2k**
**What is this feature?**

This PR adds /api/gnet to the list of ignored paths in the gzip middleware.

**Why do we need this feature?**

Without this, when gzip is enabled (`server.enable_gzip = true`), responses from the gnet proxy are double compressed: once by grafana.com and once by Grafana itself. With this change we only do one round of compression for these endpoints.

**Special notes for your reviewer:**

To test this out, try a request like this with `server.enable_gzip = true` (after setting `GCOM_TOKEN` to a valid grafana.com token; you may need to change the 'bsull' slug, too):

    curl -v --user admin:admin \
        -H "X-Api-Key: $GCOM_TOKEN" \
        -H 'Accept-Encoding: gzip' \
        localhost:3000/api/gnet/instances/bsull/provisioned-plugins/grafana-ml-app | gzip -d

Note that there are two Content-Encoding: gzip headers before this PR, and the output is still compressed even after the `gzip -d`. After this PR things look as expected.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/middleware/gziper.go
Comment (andresmgot) [inline pkg/middleware/gziper.go:48]: nit: I'd add a comment saying "already gziped by Grafana"

**Review by andresmgot — APPROVED** [review]
LGTM, tested and working as expected

File: pkg/middleware/gziper.go
Comment (sd2k) [reply pkg/middleware/gziper.go]: Good idea, done!

**Comment by sd2k** [comment]
@andresmgot could you merge this if you're happy? I don't have merge permissions on this repo 🙏

**Comment by andresmgot** [comment]
@sd2k I can but better if you get permissions to do so, you will need to merge the backports for the different branches you set. AFAIK any grafana dev should have write permissions :)

**Comment by andresmgot** [comment]
BTW, you may want to backport only to 10.0x, it depends if you consider this a "critical" bug, according to our [guides](https://internal-ops-us-east-0.grafana.net/grafana-release-guide/supported-versions/).

**Comment by sd2k** [comment]
Hmm it's blocking users from initializing the Machine Learning plugin in cloud. Most cloud instances will be running 10.x but some won't, hence the 9.5 backport. But admittedly it's probably not what I'd consider critical given that it impacts so few customers! I'll remove the backport.

**Pull request merged**
